Fixed CLI errors.

ShigrafS opened the pull request.

agalitsyna commented on pairtools/cli/sort.py:
```diff
@@ -191,16 +193,47 @@ def sort_py(
+    for col in sort_columns:
+        if col is None:
+            continue  # Skip optional columns
+        dtype = pairsam_format.DTYPES_PAIRSAM.get(column_names[col], str)
```
Looks like the user can define custom columns and their format might be not defined: https://github.com/open2c/pairtools/blob/master/pairtools/lib/pairsam_format.py

It will be great to have a specialized error message

agalitsyna commented on pairtools/cli/__init__.py:
Please, make the pairtools._logging.py file look similar to this: 
https://github.com/open2c/pairtools/pull/267/commits/f3c8f61ef9b9794aebe448727ae2c7f5a341f234

(not related to CLI but will allow the code tests to run without technical flake8 error.

ShigrafS replied:
Sure!

ShigrafS commented:
@agalitsyna I've added a warning when sorting by columns not defined in pairsam_format.DTYPES_PAIRSAM, defaulting them to string type, and updated the --extra-col help text to reflect this behavior. I've also added a test to verify warnings for undefined custom columns
The PR is ready to be merged. 
Kindly review it.

ShigrafS commented:
@golobor @agalitsyna @Phlya 
This PR is ready. Kindly review it.

agalitsyna commented on pairtools/lib/headerops.py:
```diff
@@ -22,6 +22,51 @@
 
 
 
+def canonicalize_columns(columns):
```
"canonicalize" -> "starndardize"

agalitsyna replied:
Make it a function with a single argument.

agalitsyna commented on pairtools/lib/headerops.py:
```diff
@@ -22,6 +22,51 @@
+    }
+    return [canonical_map.get(col.lower(), col) for col in columns]
+
+def get_column_index(column_names, column_spec):
```
"column_spec" -> "column/col"

agalitsyna commented on pairtools/lib/headerops.py:
```diff
@@ -22,6 +22,51 @@
+    except ValueError:
+        pass
+        
+    # Try case-insensitive
```
Remove the case-insensittive law or explain why it's needed (we have not found the usecase)

agalitsyna commented on pairtools/cli/dedup.py:
```diff
@@ -485,6 +484,34 @@ def dedup_py(
+    
+    # Get column indices with fallbacks
+    try:
+        col1 = headerops.get_column_index(column_names, c1)
```
the names like "col1/col2/cols1" are not descriptive enough, so maybe use "col_c1/col_c2/col_s1" instead: (1) use underscore, (2) preserve the recognizable name that was used before (c1/c2/p1/p2/etc.).
